packages/core-components/src/layout/Page/Page.tsx

```diff
@@ -25,7 +25,8 @@ const useStyles = makeStyles(() => ({
       "'pageHeader pageHeader pageHeader' 'pageSubheader pageSubheader pageSubheader' 'pageNav pageContent pageSidebar'",
     gridTemplateRows: 'auto auto 1fr',
     gridTemplateColumns: 'auto 1fr auto',
-    minHeight: '100vh',
+    height: '100vh',
+    overflowY: 'auto',
```

Comment:
Soooo ... I guess this is the one that leaves me pondering a bit.

Getting the entire page layout right is tricky. This PR changes it so that it isn't the outer viewport that scrolls, but instead an inner div. I'm always tempted to try to keep the scrolling as far up the tree as possible, but that's not a biggie as such - it's more that I wonder what it means for full-height type views that want to fill the viewport with a table and such. Hmmmm

I'll poke around the interface a bit and see if this affects stuff. Do you have any particular views or input on this? I'm not an expert at all and want to have a great page layout that works well for all cases, and if this does, then all is good :)

Reply:
Good catch - agree with you that it make sense to keep it up the tree & changes like this can have a non-intended impact. Nevertheless I think it make sense here, as it looks like this at the moment - stretching the sidebar when it's not necessary & preventing the user to see the full sidebar even if there would have been space:

https://user-images.githubusercontent.com/8904624/124095801-51d8b300-da5a-11eb-81b5-e95bcfae1ff0.mov

I could not confidently answer it directly, so I checked again. At the moment it's like this:

```
<body>    <-- Scrollable wrapper
  <div#root>
    <div>
      <sidebar>
      <content>
    </div>
  </div#root>
</body
```
And it would change to this: 
```
<body>    
  <div#root>
    <div>
      <sidebar>
      <content>     <-- Scrollable wrapper
    </div>
  </div#root>
</body
```

It should (in my opinion) not break things because of the following: The child of `div#root` already has a left padding reserving space for the sidebar, so overflowing content is prevented. Furthermore `body` & `div#root` still have `min-height: 100%`, so if I forgot about components rendering into the root, they should still be visible like before. Overflow in the content `div` is secured by giving it the `auto` value.

Reply:
Aight, thanks for being thorough.

I guess another take could be to have a static out-of-flow sidebar (attached to the screen edges) and give the body some left-padding. Not sure if that's necessarily better or worse, for other reasons?

Reply:
Yes - definitely! Reducing the nesting & having an out-of-flow sidebar could be achieved with a general Grid System as intended by the Material Design & MUI React. This is actually something I thought about for my larger intern task - will share my current ideas & roadmap with you. This issue here was more meant to be an easy win making it useable right now :)